frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/children/$childId/information.tsx

```diff
@@ -110,6 +110,14 @@ export async function action({ context: { session }, params, request }: ActionFu
           ctx.addIssue({ code: z.ZodIssueCode.custom, message: t('apply-adult-child:children.information.error-message.sin-required'), path: ['socialInsuranceNumber'] });
         } else if (!isValidSin(val.socialInsuranceNumber)) {
           ctx.addIssue({ code: z.ZodIssueCode.custom, message: t('apply-adult-child:children.information.error-message.sin-valid'), path: ['socialInsuranceNumber'] });
+        } else if (
+          val.socialInsuranceNumber &&
+          [applyState.applicantInformation?.socialInsuranceNumber, ...applyState.children.map((child) => child.information?.socialInsuranceNumber)]
```

Comment:
You have to exclude the current children from the children list when comparing and you also have to check with the partner information.